server/mqtt.go

```diff
@@ -4697,32 +4789,41 @@ func mqttDeliverMsgCbQoS0(sub *subscription, pc *client, _ *Account, subject, re
 		topic = pc.mqtt.pp.topic
 		// Check for service imports where subject mapping is in play.
 		if len(pc.pa.mapped) > 0 && len(pc.pa.psi) > 0 {
-			topic = natsSubjectToMQTTTopic(subject)
+			topic = natsSubjectStrToMQTTTopic(subject)
 		}
 
 	} else {
 		// Non MQTT client, could be NATS publisher, or ROUTER, etc..
 		h := mqttParsePublishNATSHeader(hdr)
 
-		// If the message does not have the MQTT header, it is not a MQTT and
-		// should be delivered here, at QOS0. If it does have the header, we
-		// need to lock the session to check the sub QoS, and then ignore the
-		// message if the Sub wants higher QOS delivery. It will be delivered by
-		// mqttDeliverMsgCbQoS12.
+		// Check the subscription's QoS. If the message was published with a
+		// QoS>0 (in the header) and the sub has the QoS>0 then the message will
+		// be delivered by mqttDeliverMsgCbQoS12.
 		if subQoS > 0 && h != nil && h.qos > 0 {
 			return
 		}
 
+		// [tck-id-conformance-mqtt-aware-nbirth-mqtt-retain] A Sparkplug Aware
+		// MQTT Server MUST make NBIRTH messages available on the topic:
+		// $sparkplug/certificates/namespace/group_id/NBIRTH/edge_node_id with
+		// the MQTT retain flag set to true
+		//
+		// [tck-id-conformance-mqtt-aware-dbirth-mqtt-retain] A Sparkplug Aware
+		// MQTT Server MUST make DBIRTH messages available on the topic:
+		// $sparkplug/certificates/namespace/group_id/DBIRTH/edge_node_id/device_id
+		// with the MQTT retain flag set to true
+		retain, _, _, _, _ = sparkbIsBirth(stringToBytes(subject), sparkbCertificatesPrefix)
```

Comment:
This will enqueue an MQTT message to the matching subscriber, but this is not going to "produce" a retained message, that is, this is not going to be stored as a retained message. Does that make sense?

Reply:
Yes, we stored it as a retained in `mqttHandlePubRetain`. Here we need to set the retained flag as per spec; the original BIRTH message must not have retain set (also per spec).

Reply:
Oh I see what you mean - we stored the retained for the BIRTH itself, but not for the BIRTH certificate. I'll review the spec once more.

Reply:
I think this reflects the spec. On a "normal" MQTT broker the BIRTH messages are not retained at all; that's the baseline. Awareness adds publishing (and retaining) the birth certificates. It makes sense that they are retained - they are supposed to contain the full set of (all) last known metrics values.